StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetVerticalViewController.swift

```diff
@@ -376,12 +407,17 @@ extension PaymentSheetVerticalViewController: VerticalPaymentMethodListViewContr
                 return nil
             }
         }()
+
+        // Hide the header label if: the only payment method type available is card and the wallet is hidden and no saved payment methods
+        let shouldHideHeader = paymentMethodTypes.count == 1 && paymentMethodType == .stripe(.card) && walletHeaderView != nil && savedPaymentMethods.isEmpty
```

Comment:
This is the form-only case, right? Rather than re-computing this logic, what about making `makeFormVC` take a `shouldHideHeader` argument that defaults to false? 

There's only one place that calls makeFormVC in the form-only case and we can pass `shouldHideHeader: walletHeaderView != nil` there.

Reply:
Yeah good idea, given this comment I think we can do this: https://stripe.slack.com/archives/C06JTLU0K8E/p1718005677330659?thread_ts=1717772333.732979&cid=C06JTLU0K8E